src/ember-app.js

```diff
@@ -384,10 +408,10 @@ class EmberApp {
 
       clearTimeout(destroyAppInstanceTimer);
 
-      if (buildSandboxPerVisit) {
+      if (buildSandboxPerVisit && !this._sandboxApplicationInstanceQueue.isFull()) {
         // if sandbox was built for this visit, then build a new sandbox for the next incoming request
         // which is invoked using buildSandboxPerVisit
-        this._pendingNewApplicationInstance = this.buildNewApplicationInstance();
+        this._sandboxApplicationInstanceQueue.enqueue(this.buildNewApplicationInstance());
```

Comment:
I'd prefer the queue to manage this state internally (avoiding the `isFull` check here).